lib/rules/selector-pseudo-class-case/index.js

```diff
@@ -28,45 +28,63 @@ const rule = function(expectation) {
       if (!isStandardSyntaxRule(rule)) {
         return;
       }
-      const selector = rule.selector;
-      const startIndexPseudo = selector.indexOf(":");
 
-      if (startIndexPseudo === -1) {
-        return;
-      }
+      rule.selectors.forEach(selector => {
+        const startIndexPseudo = selector.indexOf(":");
+
+        if (startIndexPseudo === -1) {
+          return;
+        }
+
+        parseSelector(selector, result, rule, selectorTree => {
+          selectorTree.walkPseudos(pseudoNode => {
+            const pseudo = pseudoNode.value;
+
+            if (!isStandardSyntaxSelector(pseudo)) {
+              return;
+            }
+
+            if (
+              pseudo.indexOf("::") !== -1 ||
+              keywordSets.levelOneAndTwoPseudoElements.has(
+                pseudo.toLowerCase().slice(1)
+              )
+            ) {
+              return;
+            }
+
+            const expectedPseudo =
+              expectation === "lower"
+                ? pseudo.toLowerCase()
+                : pseudo.toUpperCase();
+
+            if (pseudo === expectedPseudo) {
+              return;
+            }
+
+            if (context.fix) {
+              const fixedSelector = selector.replace(pseudo, expectedPseudo);
+
+              if (rule.raws.selector) {
+                rule.raws.selector.raw = rule.raws.selector.raw.replace(
```

Comment:
The following css will not be autofixed.

```css
.foo > /* comment */ a:Hover { color: pink; }
```

I think that this problem will not occur with the method I [proposed before](https://github.com/stylelint/stylelint/pull/3671#discussion_r219185245).
Do you have reason to process with `replace`?

I'm sorry if you can not understand, because I can not English.

Reply:
Hi, certainly not your English! :) I am just still trying to understand the postcss api, all the possible test cases, and what is the best way to solve a certain issue. Thank you for your help!